android-client-sdk/src/main/java/com/devcycle/sdk/android/api/DVCClient.kt

```diff
@@ -57,12 +57,22 @@ class DVCClient private constructor(
 
     private val configRequestQueue = ConcurrentLinkedQueue<UserAndCallback>()
     private val configRequestMutex = Mutex()
+    private val defaultCacheTTL = 7 * 24 * 3600000L // 7 days
+    private val configCacheTTL = options?.configCacheTTL ?: defaultCacheTTL
+    private val disableConfigCache = options?.disableConfigCache ?: false
 
     private var latestIdentifiedUser: PopulatedUser = user
 
     private val variableInstanceMap: MutableMap<String, MutableMap<Any, WeakReference<Variable<*>>>> = mutableMapOf()
 
     init {
+        val cachedConfig = if (disableConfigCache) null else dvcSharedPrefs.getConfig(user, configCacheTTL)
+        if (cachedConfig != null) {
```

Comment:
is `dvcSharedPrefs.getConfig() ` also checking the user id is equal to the one in cache?

Reply:
`getConfig` verifies the cached config matches the criteria (user ID and fetch date), otherwise it returns null